spec/javascripts/packages/document-capture/components/document-capture-spec.jsx

```diff
@@ -475,4 +475,90 @@ describe('document-capture/components/document-capture', () => {
 
     expect(onStepChange.callCount).to.equal(1);
   });
+
+  describe('pending promise values', () => {
+    let completeUploadAsSuccess;
+    let completeUploadAsFailure;
+    let renderResult;
+    let upload;
+    let submit;
+
+    beforeEach(async () => {
+      sandbox.stub(window, 'fetch');
+      window.fetch.withArgs('about:blank#front').returns(
+        new Promise((resolve, reject) => {
+          completeUploadAsSuccess = () => resolve({ ok: true, headers: new window.Headers() });
+          completeUploadAsFailure = () => reject(new Error());
+        }),
+      );
+      window.fetch
+        .withArgs('about:blank#back')
+        .resolves({ ok: true, headers: new window.Headers() });
+      upload = sinon.stub().resolves({ success: true, isPending: false });
+      const key = await window.crypto.subtle.generateKey(
+        {
+          name: 'AES-GCM',
+          length: 256,
+        },
+        true,
+        ['encrypt', 'decrypt'],
+      );
+      renderResult = render(
+        <UploadContextProvider
+          endpoint="about:blank#upload"
+          backgroundUploadURLs={{
+            front: 'about:blank#front',
+            back: 'about:blank#back',
+          }}
+          backgroundUploadEncryptKey={key}
+          upload={upload}
+        >
+          <ServiceProviderContext.Provider value={{ isLivenessRequired: false }}>
+            <AcuantContextProvider sdkSrc="about:blank">
+              <DocumentCapture />
+            </AcuantContextProvider>
+          </ServiceProviderContext.Provider>
+        </UploadContextProvider>,
+      );
+      const { getByLabelText, getByText } = renderResult;
+
+      userEvent.upload(getByLabelText('doc_auth.headings.document_capture_front'), validUpload);
+      userEvent.upload(getByLabelText('doc_auth.headings.document_capture_back'), validUpload);
+      submit = () => userEvent.click(getByText('forms.buttons.submit.default'));
+    });
+
+    context('success', () => {
+      it('calls to upload once pending values resolve', async () => {
+        submit();
+        expect(upload).not.to.have.been.called();
+        completeUploadAsSuccess();
+        await new Promise((resolve) => upload.callsFake(resolve));
+      });
```

Comment:
Re: Failing build, I think there's a race condition happening somewhere in here.